src/backend/distributed/connection/placement_connection.c

```diff
@@ -695,6 +738,34 @@ CanUseExistingConnection(uint32 flags, const char *userName,
 
 
 /*
+ * AccessedNonColocatedPlacement returns true if the connection accessed another
+ * placement in the same colocation group with a different representative value,
+ * meaning it's not strictly colocated.
+ */
+static bool
+AccessedNonColocatedPlacement(MultiConnection *connection, ShardPlacement *placement)
```

Comment:
Can you pick a different name ? 
This makes me think it is about deciding if the transaction had accessed any non co-located placement whereas it is about accessing a different placement of a a relation in the same co-location group.